src/main/webapp/app/pages/companyPage/CompanyAdditionalInfo.tsx

```diff
@@ -0,0 +1,142 @@
+import { CompanyAdditionalInfoDTO } from 'app/shared/api/generated/API';
+import { FormTextAreaField } from 'app/shared/textarea/FormTextAreaField';
+
+const defaultAdditionalInfo: CompanyAdditionalInfoDTO = {
+  license: {
+    autoRenewal: false,
+    activation: '',
+    termination: {
+      date: '',
+      notes: '',
+      notificationDays: 0,
+    },
+  },
+};
+
+type ICompanyAdditionalInfo = {
+  setAdditionalInfo: (
+    additionalInfo: CompanyAdditionalInfoDTO | null | undefined
+  ) => void;
+  additionalInfo: CompanyAdditionalInfoDTO | null | undefined;
+  mode: 'create' | 'update';
+};
+
+export default function CompanyAdditionalInfo({
+  additionalInfo,
+  setAdditionalInfo,
+  mode,
+}: ICompanyAdditionalInfo) {
+  const boldLabel = mode === 'update';
+
+  return (
+    <>
+      <FormInputField
+        id="activation"
+        label="Activation"
+        type="date"
+        value={additionalInfo?.license?.activation}
+        boldLabel={boldLabel}
+        onChange={event => {
+          setAdditionalInfo({
+            license: {
+              ...defaultAdditionalInfo.license,
+              ...additionalInfo?.license,
+              activation: event.target.value,
+            },
+          });
+        }}
+      />
+      <FormSelectWithLabelField
+        labelText="Auto-Renewal"
+        name="auto-renewal"
+        boldLabel={boldLabel}
+        defaultValue={{
+          value: additionalInfo?.license?.autoRenewal,
+          label: additionalInfo?.license?.autoRenewal ? 'Yes' : 'No',
+        }}
+        options={[
+          { value: true, label: 'Yes' },
+          { value: false, label: 'No' },
+        ]}
+        onSelection={autoRenewalOption => {
+          setAdditionalInfo({
+            license: {
+              ...defaultAdditionalInfo.license,
+              ...additionalInfo?.license,
+              autoRenewal: autoRenewalOption.value ?? false,
+            },
+          });
+        }}
+      />
+      <FormInputField
+        id="termination.notification-days"
+        label="Termination Notification Days"
+        type="number"
+        value={additionalInfo?.license?.termination?.notificationDays}
+        boldLabel={boldLabel}
+        onChange={event => {
+          const value = event.target.value;
+          setAdditionalInfo({
+            license: {
+              ...defaultAdditionalInfo.license,
+              ...additionalInfo?.license,
+              termination: {
+                ...defaultAdditionalInfo.license.termination,
+                ...additionalInfo?.license.termination,
+                notificationDays: value
+                  ? +value
+                  : ((undefined as unknown) as number),
+              },
+            },
+          });
+        }}
+      />
+      {mode === 'update' && (
+        <>
+          <FormInputField
+            id="termination.date"
+            label="Termination Date"
+            type="date"
```

Comment:
The newly added components seem to have different height than others?

Reply:
I was trying to copy [FormTextAreaField](https://github.com/oncokb/oncokb-public/blob/ebc63f7a4a549d0d5e426b5c43ecc2ceaa3e98aa/src/main/webapp/app/shared/textarea/FormTextAreaField.tsx) is that a bad example?

Reply:
Not sure. If this component comes from us, we need to examine why it looks differently from others.